apps/common/lib/lexical/source_file.ex

```diff
@@ -109,14 +109,18 @@ defmodule Lexical.SourceFile do
 
   def apply_content_changes(%__MODULE__{} = source, version, changes) when is_list(changes) do
     result =
-      Enum.reduce_while(changes, source, fn change, source ->
-        case apply_change(source, change) do
-          {:ok, new_source} ->
-            {:cont, new_source}
-
-          error ->
-            {:halt, error}
-        end
+      Enum.reduce_while(changes, source, fn
+        nil, source ->
```

Comment:
It really caught bugs.

Reply:
this was handled before with a function head that I removed. This seems cleaner to me.